exovetter/vetters.py

```diff
@@ -84,31 +84,46 @@ def plot(self, tce, lightcurve):
 class ModShift(BaseVetter):
     """Modshift vetter."""
 
-    def __init__(self, **kwargs):
+    def __init__(self, lc_name="flux", **kwargs):
         self.metrics = None
+        self.lc_name = lc_name
 
     def run(self, tce, lightcurve):
-        """What I want this to look like, but it doesn't work yet
         """
-        time = lightcurve.time
-        flux = lightcurve.flux
-        time_offset_str = lightcurve.time_format
+        run ModShift vetter
+        Runs modshift.compute_modeshift_metrics to populate the vetter object.
+
+        Parameters
+        -----------
+        tce : tce object
+        tce object is a dictionary that contains information about the tce
+        to vet, like period, epoch, duration, depth
+
+        lc : lightkurve object
+        lightkurve object with the time and flux to use for vetting.
+
+        """
+        self.time, self.flux, time_offset_str = \
+            lightkurve_utils.unpack_lk_version(lightcurve, self.lc_name)
+
         time_offset_q = const.string_to_offset[time_offset_str]
 
-        flux = lcutils.set_median_flux_to_zero(flux)
+        self.flux = lcutils.set_median_flux_to_zero(self.flux)
 
-        # We are content to use the epoch in the user supplied units
-        period_days = tce['period'].to_value(u.day)
-        epoch_days = tce.get_epoch(time_offset_q).to_value(u.day)
-        duration_hrs = tce['duration'].to_value(u.hour)
+        self.period_days = tce['period'].to_value(u.day)
+        self.epoch_days = tce.get_epoch(time_offset_q).to_value(u.day)
+        self.duration_hrs = tce['duration'].to_value(u.hour)
 
-        box = model.create_box_model_for_tce(tce, time * u.day, time_offset_q)
-        self.metrics = modshift.compute_modshift_metrics(
-                 time, flux, box, period_days, epoch_days, duration_hrs)
-        return self.metrics
+        self.box = model.create_box_model_for_tce(tce, self.time * u.day,
+                                                  time_offset_q)
+        self.metrics, self.conv = modshift.compute_modshift_metrics(
```

Comment:
Should this method return the metrics dict?

Are you keeping conv around for debugging? Otherwise it feels like an internal variable in the function that doesn't need to be returned.

Reply:
For all the other vetters we populate the object instead of return the values. I'm not sure it was the right decision, but it's where we are.  So consistency and all. We will need to regularize the names at some point so they make sense to anyone but us.

you are right, we don't need to keep self.conv around.